unable to compile: "cannot use func literal"

`jcdriver/output_generic.go:11:9: cannot use func literal (type func(jcpc.JoyConType, int) (jcpc.Output, error)) as type jcpc.OutputFactory in return argument`

I get the error above when I attempt to run `go get ./...` or `go get ./jcdriver` from the prog4 directory. 

I'm running OSX 10.13

Full disclosure: I'm brand new to go and am mostly fiddling to see if I can effectively use joycon with my macbook pro

--- a/prog4/jcdriver/output_generic.go
+++ b/prog4/jcdriver/output_generic.go
@@ -8,7 +8,7 @@ import (
 )
 
 func getOutputFactory() jcpc.OutputFactory {
-	return func(t jcpc.JoyConType, playerNum int) (jcpc.Output, error) {
+	return func(t jcpc.JoyConType, playerNum int, remap InputRemappingOptions) (jcpc.Output, error) {
 		return output.NewConsole(t, playerNum)
 	}
 }

